geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloFeatureIndex.scala

```diff
@@ -117,14 +123,60 @@ object AccumuloFeatureIndex extends AccumuloIndexManagerType with LazyLogging {
     }
     indices.filter(_.supports(sft))
   }
+
+  def applyVisibility(sf: SimpleFeature, key: Key): Unit = {
+    val visibility = key.getColumnVisibility
+    if (visibility.getLength > 0) {
+      SecurityUtils.setFeatureVisibility(sf, visibility.toString)
+    }
+  }
 }
 
-trait AccumuloWritableIndex extends AccumuloFeatureIndexType {
+trait AccumuloFeatureIndex extends AccumuloFeatureIndexType {
 
-  override def delete(sft: SimpleFeatureType, ds: AccumuloDataStore, shared: Boolean): Unit = {
-    import org.apache.accumulo.core.data.{Range => aRange}
-    import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+  import AccumuloFeatureIndex.{AttributeColumnFamily, BinColumnFamily, FullColumnFamily}
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+
+  /**
+    * Indicates whether the ID for each feature is serialized with the feature or in the row
+    *
+    * @return
+    */
+  def serializedWithId: Boolean
+
+  protected def hasPrecomputedBins: Boolean
+
+  override def configure(sft: SimpleFeatureType, ds: AccumuloDataStore): Unit = {
+    import scala.collection.JavaConversions._
+
+    super.configure(sft, ds)
 
+    val table = getTableName(sft.getTypeName, ds)
+
+    // create table if needed
+    AccumuloVersion.ensureTableExists(ds.connector, table)
+
+    // create splits - drop first split, otherwise we get an empty tablet
+    val splitsToAdd = getSplits(sft).map(new Text(_)).toSet -- ds.tableOps.listSplits(table).toSet
```

Comment:
i should remove that comment - it's not relevant here any more. the `getSplits` method drops the first element already. I believe the implementations do call `.tail` or `.drop(1)` or something